Sources/Sugar/Authentication/UserType.swift

```diff
@@ -36,6 +50,61 @@ extension UserType where
     }
 }
 
+extension UserType where
+    Self: Model,
+    Self: PasswordAuthenticatable,
+    Self.Database: QuerySupporting,
+    Self.Registration: HasReadableUser
+{
+    public static func preRegister(with registration: Registration, on worker: DatabaseConnectable) throws -> Future<Void> {
+        return try Self.query(on: worker)
+            .filter(Self.usernameKey == registration.username)
+            .first()
+            .nil(or: AuthenticationError.usernameAlreadyExists)
+    }
+}
+
+extension UserType where
+    Self: Model,
+    Self: PasswordAuthenticatable,
+    Self.Database: QuerySupporting,
+    Self.Update: HasUpdatableUsername,
+    Self.Update: HasUpdatablePassword,
+    Self: HasPassword
+{
+    public func preUpdate(with update: Update, on worker: DatabaseConnectable) throws -> Future<Void> {
+        if update.password != nil && (update.username == nil && update.oldPassword == nil) {
+            throw Abort(.internalServerError)
```

Comment:
Why internal server error? Isn't this an `unprocessable entity`?

Reply:
A leftover - definitely should be, thanks